app/metrics/jupyterkernelbusyness.go

```diff
@@ -0,0 +1,128 @@
+package metrics
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type JupyterKernelBusynessMetricsHandler struct {
+	logger         *logrus.Logger
+	forwardAddress string
+	listenAddress  string
+	namespace      string
+	serviceName    string
+	instanceName   string
+	metricName     MetricName
+	metric         *prometheus.GaugeVec
+}
+
+func NewJupyterKernelBusynessMetricsHandler(logger *logrus.Logger,
+	forwardAddress string,
+	listenAddress string,
+	namespace string,
+	serviceName string,
+	instanceName string) (*JupyterKernelBusynessMetricsHandler, error) {
+	return &JupyterKernelBusynessMetricsHandler{
+		logger:         logger,
+		forwardAddress: forwardAddress,
+		listenAddress:  listenAddress,
+		namespace:      namespace,
+		serviceName:    serviceName,
+		instanceName:   instanceName,
+		metricName:     JupyterKernelBusynessMetricName,
+	}, nil
+}
+
+func (n *JupyterKernelBusynessMetricsHandler) RegisterMetric() error {
+	gaugeVec := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: string(n.metricName),
+		Help: "Jupyter kernel busyness",
+	}, []string{"namespace", "service_name", "instance_name"})
+
+	if err := prometheus.Register(gaugeVec); err != nil {
+		n.logger.WithError(err).WithField("metricName", string(n.metricName)).Error("Failed to register metric")
+		return err
```

Comment:
1. Please use nuclio errors for error wrapping (context really does help)
2. Logging for every error makes the code difficult to maintain (eventually people don't log). The pattern is to wrap the error and return it and perform logging in only a couple of places (e.g. initialization and request handling)